echo-telemetry/src/test/groovy/com/netflix/spinnaker/echo/telemetry/TelemetryEventListenerSpec.groovy

```diff
@@ -367,6 +381,84 @@ class TelemetryEventListenerSpec extends Specification {
     }
   }
 
+  def "test event has deploy method information happy path"(String deploymentType, String deployVersion) {
```

Comment:
I don't think I've ever seen the test methods take parameters - are those necessary?

Reply:
Since we are using [Spock data tables](http://spockframework.org/spock/docs/1.0/data_driven_testing.html) that's why we need parameters in this test method.
The reason for the use of data tables is that we want to be sure that all the possible scenarios for [deployment Type] (https://github.com/spinnaker/kork/blob/master/kork-proto/src/main/proto/stats/log.proto) are well-tested.

Reply:
You don't need to specify the parameters - Spock will take care of that for you.